django/urls/base.py

```diff
@@ -179,3 +184,89 @@ def translate_url(url, lang_code):
             else:
                 url = urlunsplit((parsed.scheme, parsed.netloc, url, parsed.query, parsed.fragment))
     return url
+
+
+def re_path(regex, view, kwargs=None, name=None, converters=None):
+    if isinstance(view, (list, tuple)):
+        # For include(...) processing.
+        pattern = RegexPattern(regex, is_endpoint=False, converters=converters)
+        urlconf_module, app_name, namespace = view
+        return URLResolver(
+            pattern,
+            urlconf_module,
+            kwargs,
+            app_name=app_name,
+            namespace=namespace,
+        )
+    elif callable(view):
+        pattern = RegexPattern(regex, name=name, is_endpoint=True, converters=converters)
+        return URLPattern(pattern, view, kwargs, name)
+    else:
+        raise TypeError('view must be a callable or a list/tuple in the case of include().')
+
+
+def path(regex, view, kwargs=None, name=None):
+    if isinstance(view, (list, tuple)):
+        # For include(...) processing.
+        pattern = RoutePattern(regex, is_endpoint=False)
+        urlconf_module, app_name, namespace = view
+        return URLResolver(
+            pattern,
+            urlconf_module,
+            kwargs,
+            app_name=app_name,
+            namespace=namespace,
+        )
+    elif callable(view):
+        pattern = RoutePattern(regex, name=name, is_endpoint=True)
+        return URLPattern(pattern, view, kwargs, name)
+    else:
+        raise TypeError('view must be a callable or a list/tuple in the case of include().')
+
+
+def include(arg, namespace=None):
```

Comment:
I'm proposing to put these functions in `django/urls/conf.py` ("Functions for use in URLsconfs.") rather than putting more in "base.py". I started by moving `include()` in https://github.com/django/django/pull/9074. I'll rebase this PR and start updating it for the other comments if it sounds okay.

Reply:
Sounds okay.

Reply:
Agreed.

Reply:
Agreed.